src/ai/backend/test/templates/session/batch_session.py

```diff
@@ -0,0 +1,81 @@
+import asyncio
+from contextlib import asynccontextmanager as actxmgr
+from typing import AsyncIterator, override
+from uuid import UUID
+
+from ai.backend.client.session import AsyncSession
+from ai.backend.test.contexts.client_session import ClientSessionContext
+from ai.backend.test.contexts.image import ImageContext
+from ai.backend.test.contexts.session import (
+    BatchSessionContext,
+    ClusterContext,
+    CreatedSessionIDContext,
+    SessionContext,
+)
+from ai.backend.test.contexts.sse import (
+    SSEContext,
+)
+from ai.backend.test.contexts.tester import TestSpecMetaContext
+from ai.backend.test.templates.session.utils import verify_session_events
+from ai.backend.test.templates.template import (
+    WrapperTestTemplate,
+)
+
+
+class BatchSessionTemplate(WrapperTestTemplate):
+    @property
+    def name(self) -> str:
+        return "batch_session"
+
+    async def _verify_session_creation(
+        self, client_session: AsyncSession, session_name: str
+    ) -> UUID:
+        image_cfg = ImageContext.current()
+        cluster_cfg = ClusterContext.current()
+        batch_cfg = BatchSessionContext.current()
+        sess_cfg = SessionContext.current()
+        timeout = SSEContext.current().timeout
+
+        listener = asyncio.create_task(
+            verify_session_events(
+                client_session,
+                session_name,
+                timeout,
+                "session_terminated",
+                {"session_failure", "session_cancelled"},
+            )
+        )
```

Comment:
Since it was created as an asyncio task, there is still a possibility that the compute session creation work actually finishes earlier and some events are missed.
It would be good to add a task like bgtask that caches the last event in Redis for a certain period and delivers it, so could you create an issue and link it to this comment? @jopemachine